products/template/build_scripts/transform.py

```diff
@@ -37,30 +37,27 @@ def _run_dbt_step(run_args=list[str]):
 
 
 def staging():
-    _run_step("staging")
+    _invoke_dbt(["test", "--select", "source:*"])
+    _execute_sql_script("staging")
 
 
 def intermediate():
-    _run_step("libraries")
-    _run_step("green_spaces")
-    _run_step("historic_landmarks")
+    _execute_sql_script("libraries")
+    _execute_sql_script("green_spaces")
+    _execute_sql_script("historic_landmarks")
+    _invoke_dbt(["test", "--select", "libraries green_spaces historic_landmarks"])
```

Comment:
How does dbt know where to look for models? For `intermediate` step, your data contract is in `sql` dir. For `product`, it's in `product` dir.

Reply:
the [`dbt_project.yml`](https://github.com/NYCPlanning/data-engineering/blob/dm-e-to-e-product-test/products/template/dbt_project.yml#L0-L1) file tells dbt where all model files live via `model-paths: ["sql"]` ([docs](https://docs.getdbt.com/reference/project-configs/model-paths))

and two notes on how dbt knows where to look:
- 100% agree that those intermediate queries and their yml files (they may be called [`properties.yml`](https://docs.getdbt.com/reference/configs-and-properties) files) could be in a named subdirectory to group things together. I just erred on the side of not changing too much in this pass
- there are many many ways to select certain models or tests (node selection syntax  [docs](https://docs.getdbt.com/reference/node-selection/syntax)). in this case, I explicitly listed the model names. so dbt finds and runs all tests related to those model